unit-tests/run-unit-tests.py

```diff
@@ -264,79 +267,111 @@ def check_log_for_fails(log, testname, exe):
+        self.executable = executable
+        self.log = logdir + '/' + testname + '.log'
 
-    progress( testname, '>', log, '...' )
-    n_tests += 1
-    try:
-        run( [exe], stdout=log )
-    except FileNotFoundError:
-        error( red + testname + reset + ': executable not found! (' + exe + ')' )
-        continue
-    except subprocess.CalledProcessError as cpe:
-        if not check_log_for_fails(log, testname, exe):
-            # An unexpected error occurred
-            error( red + testname + reset + ': exited with non-zero value! (' + str(cpe.returncode) + ')' )
-
-# If we run python tests with no .pyd/.so file they will crash. Therefore we only run them if such a file was found
-if pyrs:
-    # unit-test scripts are in the same directory as this script
-    for py_test in find(current_dir, '(^|/)test-.*\.py'):
-    
-        testdir = py_test[:-3]                         # "log/internal/test-all"  <-  "log/internal/test-all.py"
-        testparent = os.path.dirname(testdir)          # same as for cpp files
-        if testparent:
-            testname = 'test-' + testparent.replace( '/', '-' ) + '-' + os.path.basename(testdir)[5:]
-        else:
-            testname = testdir
-    
-        log = logdir + '/' + testname + '.log'
-        
-        progress( testname, '>', log, '...' )
+    def run_test(self):
+        global n_tests
+        progress(self.testname, '>', self.log, '...')
         n_tests += 1
-        test_path = current_dir + os.sep + py_test
-        if linux:
-            cmd = ["python3", test_path]
-        else:
-            cmd = ["py","-3", test_path]
         try:
-            run( cmd, stdout=log )
+            run( self.command, stdout=self.log )
         except FileNotFoundError:
-            error( red + testname + reset + ': file not found! (' + test_path + ')' )
-            continue
+            error(red + self.testname + reset + ': executable not found! (' + self.executable + ')')
         except subprocess.CalledProcessError as cpe:
-            if not check_log_for_fails(log, testname, py_test):
+            if not check_log_for_fails(self.log, self.testname, self.executable):
                 # An unexpected error occurred
-                cat(log)
-                error(red + testname + reset + ': exited with non-zero value! (' + str(cpe.returncode) + ')')
+                error(red + self.testname + reset + ': exited with non-zero value! (' + str(cpe.returncode) + ')')
+
+class PyTest(Test):
+    """
+    Class for python tests
+    """
+    def __init__(self, testname, path_to_test):
+        """
+        :param testname: name of the test
+        :param path_to_test: the relative path from the current directory to the path
+        """
+        global current_dir
+
+        if linux:
+            cmd = ["python3", path_to_test]
+        else:
+            cmd = ["py", "-3", path_to_test]
+
+        Test.__init__(self, testname, cmd, current_dir + path_to_test)
+
+class CTest(Test):
```

Comment:
Not sure the language is important... I.e., it could be a compiled Matlab test, too. Point is that it's an executable, right?